src/EFCore.MySql/MySqlDbFunctionsExtensions.cs

```diff
@@ -450,5 +450,54 @@ public static class MySqlDbFunctionsExtensions
             => (startDate.HasValue && endDate.HasValue)
                 ? (int?)DateDiffMicrosecond(_, startDate.Value, endDate.Value)
                 : null;
+
+        /// <summary>
+        ///     <para>
+        ///         An implementation of the SQL LIKE operation. On relational databases this is usually directly
+        ///         translated to SQL.
+        ///     </para>
+        ///     <para>
+        ///         Note that if this function is translated into SQL, then the semantics of the comparison will
+        ///         depend on the database configuration. In particular, it may be either case-sensitive or
+        ///         case-insensitive. If this function is evaluated on the client, then it will always use
+        ///         a case-insensitive comparison.
+        ///     </para>
+        /// </summary>
+        /// <param name="_">The DbFunctions instance.</param>
+        /// <param name="matchExpression">The property of entity that is to be matched.</param>
+        /// <param name="pattern">The pattern which may involve wildcards %,_,[,],^.</param>
+        /// <returns>true if there is a match.</returns>
+        public static bool Like<T>(
+            [CanBeNull] this DbFunctions _,
+            T matchExpression,
```

Comment:
It appears (from https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/pull/806/files#diff-8764c7d56fa4b46141c410f9975704ebR63) that this can be specialised on `double`, `decimal`, etc. `ToString` is not safe to call in non-U.S.-English cultures, because it can format `1.23` as `"1,23"`, which won't be accepted by MySQL Server. 

Can `T` here be a type that needs culture-invariant formatting? If so, you might need to use `IConvertible`.

Reply:
I have already used `IConvertible` in the latest commit.